### PR TITLE
Json schema 2 po ph po

### PR DESCRIPTION
**Pull request by bscSCORM**
Re-created #10 from @jwisener 
- **Added generator to generate pure PHP objects.** 

```
public class PhpCodeDomProvider : CodeDomProvider, ICodeGenerator
    {
        ICodeGenerator _generator;
        public PhpCodeDomProvider()
        {
            _generator = new PhpCodeGenerator();
        }
        [Obsolete("Callers should not use the ICodeGenerator interface and should instead use the methods directly on the CodeDomProvider class. Those inheriting from CodeDomProvider must still implement this interface, and should exclude this warning or also obsolete this method.")]
        public override ICodeGenerator CreateGenerator()
        {
            return _generator;
        }
```
- **The majority of the work was mainly done within this class.** 

```
 public class PhpCodeGenerator : ICodeGenerator
{
    ...
}
```
- **Also note a new -l flag was added to the command line arguments so we know which codeprovider to use during the generations.**

```
{"l=|language=","Language to export (csharp or php)",  ln  => settings.Config.LanguageExportType = (LanguageExportType)Enum.Parse(typeof(LanguageExportType),ln,true)},
                 {
```

**\- The way the code was tested was by running within visual studo.net and passing it both the -l csharp and another run passing it the -l php flag. It generated the pure language object respectively.**

### CODE REVIEW

File: Source/Cvent.SchemaToPoco.Console/JsonSchemaToPoPHPoCLExecutor.cs [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+    /// <summary>
+    ///     Main executor class for the CLI.
+    /// </summary>
+    // ReSharper disable once InconsistentNaming
+    public class JsonSchemaToPoPHPoCLExecutor
+    {
+        /// <summary>
+        ///     Main controller.
+        /// </summary>
+        private readonly JsonSchemaToPoPHPo _controller;
```
Comment (bscSCORM) [inline Source/Cvent.SchemaToPoco.Console/JsonSchemaToPoPHPoCLExecutor.cs]: FYI, you don't need to comment private members generally

**Comment by bscSCORM** [comment]
the schematopophppo seems to have a missing dependency on PHP.Core.CodeDom, also it looked like you added an extra main console class / project but then also talk about being able to switch between languages on the command line, so let’s just talk through this tomorrow

**Comment by jwisener** [comment]
Ah, sorry about that! that Dependency was not needed. I was trying something with that library when starting on this originally, only to be quite disappointed as the library would not handle enums properly (among other things). I have removed that dependency.

**Comment by jwisener** [comment]
Yes, the  extra main console class / project was left over as well, I was working in visual studio and it wasn't showing me those files. Originally, I thought, I'd would create a whole new executable/project, but then as I studied the existing code more, they already had it designed to "extend" for extra providers. Thus I went with that and forgot to remove those two files (I have removed both that project file and the JsonSchemaToPoPHPoCLExecutor.cs as those are not needed).

**Comment by bscSCORM** [comment]
👍 @kpatmor 
note, the tests don't work w/o manually bringing down nunit tools, but that's the case on master as well

**Comment by bscSCORM** [comment]
@kpatmor hold off on merging @jwisener is going to make this default to c# so we don't have to change other places that are calling it.

**Comment by bscSCORM** [comment]
@kpatmor now I'm 👍 to merge

**Pull request merged**